service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -193,7 +194,17 @@ void testUpdateMetadataWithInvalidUser() {
   @Test
   void testUpdateMetadata() {
     mockDataset();
-    String metadata = "test metadata";
+    String metadata = """
+        {"test": "metadata"}""";
+    when(samService.hasGlobalAction(user, SamAction.UPDATE_ANY_METADATA)).thenReturn(true);
+    datasetService.updateMetadata(user, datasetId, metadata);
+    verify(datasetDao).update(dataset.withMetadata(metadata));
+  }
+
+  @Test(expected = BadRequestException.class)
+  void testUpdateMetadataInvalidInput() {
+    mockDataset();
+    String metadata = "metadata must be json object";
     when(samService.hasGlobalAction(user, SamAction.UPDATE_ANY_METADATA)).thenReturn(true);
     datasetService.updateMetadata(user, datasetId, metadata);
     verify(datasetDao).update(dataset.withMetadata(metadata));
```

Comment:
Does this work? I'd expect that the dao is never called. In fact you may want to assert that using `verifyNoInteractions()`.